app/src/main/java/cmput301f17t13/com/catisadog/activities/summary/ViewHabitActivity.java

```diff
@@ -1,36 +1,134 @@
+ * law or agreed to in writing, software.
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  */
 
 package cmput301f17t13.com.catisadog.activities.summary;
 
-import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.text.TextUtils;
 import android.view.View;
+import android.widget.TextView;
+
+import org.joda.time.DateTimeConstants;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+import cmput301f17t13.com.catisadog.R;
+import cmput301f17t13.com.catisadog.models.Habit;
+import cmput301f17t13.com.catisadog.utils.IntentConstants;
 
 /**
  * A screen for viewing habit details
  *
  * @see cmput301f17t13.com.catisadog.models.Habit
  */
-public class ViewHabitActivity extends Activity {
+public class ViewHabitActivity extends AppCompatActivity {
+    private Habit habit;
+
+    /**
+     * OnCreate method to setup the UI and fill TextView values with habit data
+     * @param savedInstanceState
+     */
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Intent intent = getIntent();
+        Bundle bundle = intent.getExtras();
+        habit = (Habit) bundle.getSerializable(
+                IntentConstants.VIEW_HABIT_INTENT_DATA);
+        setContentView(R.layout.activity_view_habit);
+        ((TextView) findViewById(R.id.habitTitle)).setText(habit.getTitle());
+        ((TextView) findViewById(R.id.habitReason)).setText(habit.getReason());
+        ((TextView) findViewById(R.id.habitStartValue)).setText(startsText());
+        ((TextView) findViewById(R.id.habitRepeatsValue))
+                .setText(repeatsText());
+        ((TextView) findViewById(R.id.habitCompletionMetricsValue))
+                .setText(completionMetricsText());
+    }
 
     /**
      * Navigate to edit the habit if it belongs to the current user
      * @param v the button view
      */
     public void editHabit(View v) {
+        Bundle bundle = new Bundle();
+        bundle.putSerializable(IntentConstants.EDIT_HABIT_INTENT_DATA, habit);
+        Intent intent = new Intent(this, ViewHabitActivity.class);
+        intent.putExtras(bundle);
+        startActivity(intent);
+    }
 
+    /**
+     * Get formatted string for start text TextView
+     * @return String representation of start text
+     */
+    private String startsText() {
+        return habit.getStartDate().toString("EEEE MMMM dd, yyyy");
     }
 
     /**
-     * Delete the habit if it belongs to the current user
-     * @param v the button view
+     * Get formatted string for repeats TextView
+     * Speial formatting if every day, weekends, or weekdays are selected
+     * @return String representation of repeat text
      */
-    public void deleteHabit(View v) {
+    private String repeatsText() {
```

Comment:
Uh, ok.

Reply:
Bad diff. I'm removing the existing (and empty) `deleteHabit` method since we agreed it should go in `EditHabit`. I added a separate method to return the text for the repeats TextView.

Reply:
I was mostly referring to the bloated nature of the function you wrote (repeatsText). Should have been more clear. Anyway, not a functional issue so feel free to merge.

Reply:
I agree that it feels a bit bloated. This was the cleanest way to implement the functionality given the libraries available and existing data structures. Cleaner implementations would require a refactor of how we get/store the dates.